staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -87,12 +87,18 @@ func WithAudit(handler http.Handler, requestContextMapper request.RequestContext
 			return
 		}
 
+		// send first event for non-read-only requests
+		ri, _ := request.RequestInfoFrom(ctx)
+		if !(ri != nil && ri.ReadOnly) {
```

Comment:
For the option, I meant something for all requests, not just read-only requests. But in either case, we can discuss in a follow up.